File: C++/54-字符流中第一个不重复的字符.cpp
class Solution
{
public:
    queue<int> q;
    int cnt[256];
    //Insert one char from stringstream
    void Insert(char ch)
    {
        cnt[ch]++;
        if (cnt[ch] == 1)
            q.push(ch);
    }
    //return the first appearence once char in current stringstream
    char FirstAppearingOnce()
    {
        while (q.size() > 0 && cnt[q.front()] > 1)
            q.pop();
        if (q.size() == 0)
            return '#';
        return q.front();
    }
};

File: Python/54-字符流中第一个不重复的字符.py
# -*- coding:utf-8 -*-
class Solution:
    # 返回对应char
    def __init__(self):
        self.ch = [0] * 256
        self.q = []
    def FirstAppearingOnce(self):
        # write code here
        while self.q and self.ch[ord(self.q[0])] > 1:
            self.q.pop(0)
        if not self.q:
            return '#'
        return self.q[0]
    def Insert(self, char):
        # write code here
        self.ch[ord(char)] += 1
        if self.ch[ord(char)] == 1:
            self.q.append(char)